nservicebus/messaging/routing.md

```diff
@@ -67,3 +67,32 @@ In the `UnicastBusConfig/MessageEndpointMappings` configuration section, publish
 ## Reply routing
 
 Reply message are always routed based on the `ReplyTo` header of the initial message regardless of the endpoint's routing configuration. Only the sender of the initial message can influence the routing of a reply. Refer to documentation on [sending messages](/nservicebus/messaging/send-a-message.md) for further details.
+
+
+
+## Make instance uniquely addressable
+
+Note: It is not advised to not have routing logic to address a specific instance be part of different logical sender as this tighly couples the sender its sending logic to the reciepient its deployment configuration.
+
+When using a message broker, multiple instances of a scaled-out endpoint are consuming from the same address via the [competing consumer](/nservicebus/architecture/scaling.md#scaling-out-to-multiple-nodes-competing-consumers) model.
+
+To address specific instances of a scaled-out endpoint, instances can be configured to be individually addressable by providing a unique discriminator to each instance:
+
+The following queues will be created for endpoint `Sales` configured with discriminator `B`:
+
+- `Sales`
+- `Sales-B`
+
+```c#
+var endpointConfiguration = new EndpointConfiguration("Sales");
+endpointConfiguration.MakeInstanceUniquelyAddressable("B");
+```
+
+Uniquely addressable instances are used for [callbacks](/nservicebus/messaging/callbacks.md) but can be used for other purposes like data partitioning with processing affinity or a form or processing prioritization.
+
+```c#
+var options = new SendOptions();
+options.RouteToThisInstance()
+options.RouteToSpecificInstance("B");
+endpointInstance.Send(new MyMessage(), options);
+```
```

Comment:
```suggestion
```

Recommendations:

- Avoid hard coding the discriminator for sending messages.
- Avoid sending to a specific instances of a different logical endpoint to prevent coupling.
- Avoid using `MakeInstanceUniquelyAddressable` for priority queues

```